lumicks/pylake/kymotracker/kymotrack.py

```diff
@@ -428,6 +428,11 @@ class KymoTrackGroup:
 
     def __init__(self, kymo_tracks):
         self._src = kymo_tracks
+        if self._src:
+            n_kymos = len(set([track._kymo._id for track in kymo_tracks]))
+            assert n_kymos == 1, "All tracks must have the same source kymograph."
+            n_channels = len(set([track._channel for track in kymo_tracks]))
+            assert n_channels == 1, "All tracks must be from the same color channel."
```

Comment:
Would it make sense to have one central function, which could be called from here and further down (line 504-512)? This would deduplicate some of the code to check if kymos are compatible.
```
def validate_compatibility(self, kymo_tracks, compare_to_self=False):
    new_kymos = set([track._kymo._id for track in kymo_tracks])
    assert (
        len(new_kymos) == 1 and (not compare_to_self or self._kymo._id == list(new_kymos)[0])
    ), "All tracks must have the same source kymograph."
    new_channels = set([track._channel for track in other])
    assert (
        len(new_channels) == 1 and (not compare_to_self or self._channel == list(new_channels)[0])
    ), "All tracks must be from the same color channel."
```

Reply:
good idea, I switched it up a bit from your suggestion because I find that negating things tends to be more confusing than using the ternary